[rpc] Descriptions: Consistent arg labels for types 'object', 'array', 'boolean', and 'string'

ch4ot1c opened the pull request:
Followup for https://github.com/bitcoin/bitcoin/pull/14459. See `scripted-diff:` commit messages for scripts used.

ch4ot1c committed:
scripted-diff: [rpc] Descriptions: Consistent arg labels for types 'object' and 'boolean'

-BEGIN VERIFY SCRIPT-

git grep -l "(bool) " src | xargs sed -i "s/(bool) /(boolean) /g"
git grep -l "(bool, " src/rpc | xargs sed -i "s/(bool, /(boolean, /g"
git grep -l "(bool, " src/wallet | xargs sed -i "s/(bool, /(boolean, /g"

git grep -l "(json object) " src | xargs sed -i "s/(json object) /(object) /g"
git grep -l "(json object, " src | xargs sed -i "s/(json object, /(object, /g"

git grep -l "(json array" src | xargs sed -i "s/(json array/(array/g"
git grep -l "of json objects" src | xargs sed -i "s/of json objects/of objects/g"

-END VERIFY SCRIPT-

MarcoFalke commented on src/rpc/mining.cpp:
```diff
@@ -878,23 +878,23 @@ static UniValue estimaterawfee(const JSONRPCRequest& request)
             "      },\n"
-            "      \"fail\" : { ... },       (json object, optional) information about the highest range of feerates to fail to meet the threshold\n"
+            "      \"fail\" : { ... },       (object, optional) information about the highest range of feerates to fail to meet the threshold\n"
             "      \"errors\":  [ str... ]   (json array of strings, optional) Errors encountered during processing\n"
```
Could remove the json here as well?

maflcko reviewed (commented):
Concept ACK on making this consistent. Later improvements to make the documentation machine generated would be easier to review.

I don't have an opinion on the actual type name, but it should be consistent.

ch4ot1c committed:
scripted-diff: [rpc] Descriptions: Pluralize 'string' when used in 'array'

-BEGIN VERIFY SCRIPT-

git grep -l "of string)" src | xargs sed -i "s/of string)/of strings)/g"

-END VERIFY SCRIPT-

MarcoFalke commented:
Needs rebase, but should be simple for a scripted-diff. Also note that RPCHelpMan uses "json array" and "json object", as opposed to just "object"/"array".

ch4ot1c commented:
Closing in favor of #14459.
